key.ui/src/main/java/de/uka/ilkd/key/ui/MediatorProofControl.java

```diff
@@ -95,6 +95,9 @@ public void stopAutoMode() {
      */
     @Override
     public void waitWhileAutoMode() {
+        if (SwingUtilities.isEventDispatchThread()) {
+            return; // do not block the UI thread
```

Comment:
This is really unexpected. Maybe this should throw an exception instead of simply returning? Blocking the event thread is a programming error.

Reply:
I think the only usage on the event thread is in `AutoModeAction`. I will change it there.